lib/gds-components/error-summary/ErrorSummary.tsx

```diff
@@ -0,0 +1,34 @@
+/**
+ * GDS component: https://design-system.service.gov.uk/components/error-summary/
+ */
+
+import { ValidationErrorItem } from "joi";
+
+interface ErrorSummaryProps {
+  errors: ValidationErrorItem[];
+  title?: string;
+}
+
+export const ErrorSummary = ({
+  title = "There is a problem",
+  errors,
+}: ErrorSummaryProps) => {
+  if (!errors.length) return null;
+
+  return (
+    <div className="govuk-error-summary" data-module="govuk-error-summary">
+      <div role="alert">
+        <h2 className="govuk-error-summary__title">{title}</h2>
+        <div className="govuk-error-summary__body">
+          <ul className="govuk-list govuk-error-summary__list">
+            {errors.map((error) => (
+              <li>
+                <a href={`#${error.path.join("-")}`}>{error.message}</a>
```

Comment:
FYI, I don't think this works in Storybook but I think it should be fine in Pages. Raising this as a note that although it's fine for now, we might have to look at this again later depending on how we end up deploying this form in the future.